### PR TITLE
Big ol' refactor of deploy logic

### LINKED ISSUE #23: Fix pip upgrade path when version stays constant but url changes
Tricky to describe but here it goes... in certain scenarios we have had to fork an upstream python package and make some changes only to have those changes later be merged in upstream. Some of the times when the packager does this, they do not bump version numbers even when they add additional features on top of our changes. This is problematic in the current pip upgrade logic as it doesnt see the version numbers as changing so we need to either experiment with using something like `pip-sync` or trying more drastic over-ride measures. Or maybe move to some kind of host packaging deployment method (as many other python ops people recommend -- ie using deb packages ).

### PR DESCRIPTION
**Pull request by msheiny**
Fixes #23, #18 so far -- will have to re-evaluate some others. What does this PR change? 

So - currently in `master`, the role logic involves remoting into a prod server and installing all of the pre-reqs and making subsequent changes. This has caused problems in production, especially on a live site and when pip requirements change in weird ways. So the changes introduced in this PR add two big things:

* All of the build logic is conducted locally under docker containers -- including pulling site code, virtualenv building, node installation. This means less dependencies installed on the server.
* We add parallel set of gunicorn services that toggle back and forth between which one is active (basically pointing the reverse-proxy, in our case `nginx` to one service or the other at the conclusion of every deploy).

A few things to keep in mind:
* This PR breaks backwards compatibility in a big way. So any previous deploys we did using this role will basically have to have the servers rebuilt. This is fine because we are using ansible so those servers aren't snowflakes :)
* The deploys will take slightly longer because whenever there is a code change we are now building pip requirements and node logic from scratch instead of building upon what was already on the server. Ultimately this is the right call -- and I tried to make the role smart enough to only do that when there is a code change (idempotency seems to be working fine).

Right right so you can test with:
* Install all the reqs under `devops/requirements.txt` into a virtualenv
* Run `make`

I'm throwing up this PR a little early to get some feedback though since its such a breaking change. I suspect there are a few dangling default variables that arent be used anymore. I'll try to clean this up as well.

### CODE REVIEW

**Comment by conorsch** [comment]
Finally sitting down to review this monster. Molecule v2 integration looks great!

File: README.md [hunk 1]
```diff
@@ -132,29 +151,34 @@ django_stack_force_refresh: false
...
+In order to run a test of the role itself:
+
+* Install pip requirements in `devops/requirements.txt` (recommend using a
+  virtualenv)
+* Run `make ci-go`
 
 If you want to experiment with your own site repo, you'll either have to start
 folding this role into your deployment playbook and experiment there or start
-making local edits to `devops/playbook.yml` accordingly.
+making local edits to `molecule/cis/playbook.yml` accordingly.
```
Comment (conorsch) [inline README.md]: Nit: `s/cis/ci/`

File: defaults/main.yml [hunk 2]
```diff
@@ -45,12 +53,13 @@ django_stack_es_host_url: http://localhost:9200
 django_stack_es_ca_path: /etc/ssl/certs/testca_freedom_press.pem
 
 #npm settings
-django_stack_npm_install_cmd: npm install
+django_stack_npm_install_cmd: "npm install; npm run start"
```
Comment (conorsch) [inline defaults/main.yml:56]: Shouldn't the chaining logic here use `&&` rather than `;`?

File: tasks/npm.yml [hunk 3]
```diff
@@ -1,48 +1,14 @@
...
-  file:
-    state: directory
-    path: "{{ django_stack_npm_dir }}"
-    recurse: yes
-    owner: "{{ django_stack_gcorn_user }}"
-    group: "{{ django_stack_gcorn_group }}"
-  changed_when: false
+- name: Establish some npm specific vars for current play
+  set_fact:
+    npm_touch: "touch /status/npm.done"
```
Comment (conorsch) [inline tasks/npm.yml]: This tempfile logic is difficult to understand. Some comments would be warranted. Near as I can figure, this task is referencing a static location incorrectly, when it should be using `"{{ tmp_dir }}/npm.done" instead. More explicit var names would help here, e.g. `django_stack_local_tmp_dir`.

Comment (conorsch) [reply tasks/npm.yml]: Ah, I see: the `/status` location is hardcoded because it's mounted via volumes. Take this a nit for comments, then. I'll add!

**Comment by conorsch** [comment]
Ingenious use of tooling to get more reliable deployments, @msheiny. You had to jump through a lot of hoops you wouldn't've had to if were deploying with containers directly, but, well, _one day_ :wistful_sigh:.

Addressed my own nits during review, and reorged the task flow a bit. I also added a cleanup task to remove the helper containers on the Ansible controller after the role run completes.

**Review by conorsch — APPROVED** [review]
In general I have concerns about the liberal use of `set_fact`, but the role has become quite dynamic, so there's not really a better way. Top job. 

During review I did _not_ rename all the `tmp_dir` vars, but I was certainly tempted to. I'll just leave this here: https://a-nickels-worth.blogspot.com/2016/04/a-guide-to-naming-variables.html

**Comment by msheiny** [comment]
I concede it is a confusing mess of variables -- i swear ill doing a second sweep of var cleanup when I add more tests in #26  :pinky: . Thanks for your kind words and sweet review skills ❤️

File: defaults/main.yml (hunk 2 above)
Comment (msheiny) [reply defaults/main.yml:56]: i honestly dont think it matters -- ive had a script bomb out before here and it just kills the entire container and the play stops with an error. im hesitant to change it though -- its running in `ash` , not `bash` under a alpine linux container

**Pull request merged**